### ISSUE TITLE
Proxmox 8.0.X support

### ISSUE THREAD

**Issue by gertz22** [theforeman/foreman_fog_proxmox#283]
Hello I want to ask about Proxmox 8.0.X support? looks like Proxmox version Unknown is not semver suitable.
Thank you.

**Comment by phantomdez** [comment]
I am guessing this is why I cannot get a listing of VMs or templates after setting up the compute resource.   I was able to get a successful connection but couldn't associate VMs

**Comment by dsexton18** [comment]
> Hello I want to ask about Proxmox 8.0.X support? looks like Proxmox version Unknown is not semver suitable. Thank you.

I get the same results. Unable to make to get the plugin to work with proxmox 8 .

**Comment by Manisha15** [comment]
> > Hello I want to ask about Proxmox 8.0.X support? looks like Proxmox version Unknown is not semver suitable. Thank you.
> 
> I get the same results. Unable to make to get the plugin to work with proxmox 8 .

Which version of the plugin are you using? Is it still throwing error with 0.14.3 version?

**Comment by dsexton18** [comment]
> > > Hello I want to ask about Proxmox 8.0.X support? looks like Proxmox version Unknown is not semver suitable. Thank you.
> > 
> > 
> > I get the same results. Unable to make to get the plugin to work with proxmox 8 .
> 
> Which version of the plugin are you using? Is it still throwing error with 0.14.3 version?

After upgrading to Foreman 3.8 and Katello 4.10 from 3.7 I was able to add my proxmox 8.0 server as a computer resource. So far it appears to work. I have not done much testing but it looks promising.  Thanks for making my day.

**Comment by Toutzn** [comment]
Yes, but not all CPU types are in the list. Also, if I select 2 sockets and 2 cores, in Proxmox it is placed with 1 socket and 2 cores.

Need all CPUs and Supported Kernel for FreeBSD. FreeBSD will work also with the Linux Kernel Config in ProxMox.

<img width="658" alt="image" src="https://github.com/theforeman/foreman_fog_proxmox/assets/3788105/ca4f6e7d-fc9c-4c48-b2a9-ea0b97848ace">

<img width="598" alt="image" src="https://github.com/theforeman/foreman_fog_proxmox/assets/3788105/114f756d-a9ac-471c-9171-feb8e47e8789">


`
Running OS: Ubuntu 20.04LTS
Foreman 3.9.1
foreman_fog_proxmox 0.15.0
`

**Comment by mfstanek** [comment]
has anyone had any luck creating a VM on Proxmox 8.2?  
I keep getting this error:

`2024-05-16T22:23:45 [E|app|ad2d9ebf] Failed to save: Failed to create a compute PROXMOX (Proxmox) instance lela-longford.example.com: Expected([200]) <=> Actual(400 Bad Request)  
ad2d9ebf | excon.error.response  
ad2d9ebf | :body => "{"data":null,"errors":{"ostype":"value 'debian' does not have a value in the enumeration 'other, wxp, w2k, w2k3, w2k8, wvista, win7, win8, win10, win11, l24, l26, solaris'","arch":"value 'amd64' does not have a value in the enumeration 'x86_64, aarch64'"}}" 
... 
ad2d9ebf | :reason_phrase => "Parameter verification failed." `

using Foreman 3.10 and foreman_fog_proxmox 0.15.0

**Comment by Kariton** [comment]
> > > Hello I want to ask about Proxmox 8.0.X support? looks like Proxmox version Unknown is not semver suitable. Thank you.
> > 
> > 
> > I get the same results. Unable to make to get the plugin to work with proxmox 8 .
> 
> Which version of the plugin are you using? Is it still throwing error with 0.14.3 version?

I think i got that issue when using auth token instead of password credentials.
need to recheck. 
creds did work.

**Comment by Kariton** [comment]
> has anyone had any luck creating a VM on Proxmox 8.2? I keep getting this error:
> 
> `2024-05-16T22:23:45 [E|app|ad2d9ebf] Failed to save: Failed to create a compute PROXMOX (Proxmox) instance lela-longford.example.com: Expected([200]) <=> Actual(400 Bad Request) ad2d9ebf | excon.error.response ad2d9ebf | :body => "{"data":null,"errors":{"ostype":"value 'debian' does not have a value in the enumeration 'other, wxp, w2k, w2k3, w2k8, wvista, win7, win8, win10, win11, l24, l26, solaris'","arch":"value 'amd64' does not have a value in the enumeration 'x86_64, aarch64'"}}" ... ad2d9ebf | :reason_phrase => "Parameter verification failed." `
> 
> using Foreman 3.10 and foreman_fog_proxmox 0.15.0

https://community.theforeman.org/t/ssl-error-on-create-image/37947/6

this helped me.
especially: 
```
Deployment of hosts does not as it seems Proxmox API has significant changes.

Dirty workaround again

Edited foreman_fog_proxmox-0.15.0/app/helpers/proxmox_compute_selectors_helper.rb
Modified “def proxmox_archs_map” first line to read:

OpenStruct.new(id: 'x86_64', name: '64 bits')

NB! i386 is not even supported by API any more. And “arch” can only be set by root@pve. So I changed user at compute resource proxmox connection

Modified all def proxmox_ostypes_map except “Unmanaged” to read:

OpenStruct.new(id: ‘l26’, name: ‘Debian’)

NB! name: ‘Debian’ is only example. The workaround is id: ‘l26’.
```

"l26" is the new supported "os type" for linux.
i interpret those values as "linux 2.4" and "linux > 2.6" eg. "6.x - 2.6 Kernel / 2.4 Kernel"

**Comment by Rascalov** [comment]
Sharing my (limited) experience and wondering how others are holding up. Anyone got through Host creation with working VM, enrollment etc.? 

Setup: I run a 3 node cluster on version 8.4.1 with ceph.
So far, I managed to create hosts. But the resulting VMs are unusable and don't use the provided image, which might be my setup as I've never done proxmox provisioning through foreman. 

I set up a template from a image created with packer. 
I add that image to the compute resource (which wrongly shows 3 images, which is probably due to ceph supplying the image on all nodes). 
![Image](https://github.com/user-attachments/assets/a41ebf53-fe7b-4b0f-8296-f73ed75f92d7)

I set up a basic compute profile and set the added image (but whenever I go back, the image value is blank again, not sure what is if/what is inserted). 

![Image](https://github.com/user-attachments/assets/a7acad87-a637-4fa4-938d-618f8118e4e2)

Then I realized I can only create the Host on the Cluster node that currently has the template (in my case pve3), otherwise it will throw a 'volid' on nil object error on submitting the Host creation form.

![Image](https://github.com/user-attachments/assets/fd9726ed-e997-41c4-bdab-0160f6933cbe)

I do an image based provision where I select the image and base proxmox user-data template.

![Image](https://github.com/user-attachments/assets/63702880-2887-4db0-a6e3-0eb56aec67d1)
 
It creates the VM and adds the host to foreman, but upon looking at the VM console, I see that it has no valid boot disk. 

![Image](https://github.com/user-attachments/assets/397517bb-5f35-47bf-a1fe-68ac9b15f6b7)

Looking at the VM hardware, it has an empty 8GB disk and the template's disk unused. Adding the unused disk to the VM has no effect.  In https://github.com/theforeman/foreman_fog_proxmox/issues/391 it was pointed out that they had to remove the hard disk from provisioning. But as far as I see, you can't remove it, as it will just add itself back by default (both in the compute profile and the Host creation). 

![Image](https://github.com/user-attachments/assets/2fb7f68f-ecfb-4927-a332-fd744d92698a)


I hope that I just have a fundamental misunderstanding on how this works, do let me know, I might make some separate issues on the cluster specific issues I faced (template-image shown thrice, only able to provision on node that has the template image, etc.)

**Comment by Manisha15** [comment]
Proxmox 8 and 9 are supported by plugin. Please create separate issue if you have specific error.
